src/index.ts

```diff
@@ -4,60 +4,53 @@ import {
   DynamoDBClientConfig
 } from '@aws-sdk/client-dynamodb';
 import { DynamoDBDocument } from '@aws-sdk/lib-dynamodb';
-
-import { execute, retryUnprocessedItems } from './dynamodb/batch/execute';
 import { batchGet } from './dynamodb/batch/get';
+
 import { batchRemove } from './dynamodb/batch/remove';
 import { batchWrite } from './dynamodb/batch/write';
 import { getItem } from './dynamodb/get-item';
 import { putItem } from './dynamodb/put-item';
 import { query, queryByIndex } from './dynamodb/query';
 import { remove } from './dynamodb/remove';
 import { scan } from './dynamodb/scan';
-import { addXrayTraceId } from './middlewares/add-xray-trace-id';
-import { removeXrayTraceId } from './middlewares/remove-xray-trace-id';
 import {
   BaseFetchOptions,
   BaseOptions,
   BaseSaveOptions,
   MhDynamoClientOptions
 } from './types';
-import {
-  ensureValid,
-  ensureValidBase,
-  ensureValidBatch,
-  ensureValidBatchWrite,
-  ensureValidQuery
-} from './validation';
 
 export class MhDynamoClient {
   private globalOptions: MhDynamoClientOptions;
   protected documentClient: DynamoDBDocument;
 
   static fromClient(client: DynamoDBClient, options?: MhDynamoClientOptions) {
-    return new MhDynamoClient(client, options);
+    return new MhDynamoClient(client, undefined, options);
   }
 
   static fromConfig(
-    dynamoDbClientConfig: DynamoDBClientConfig,
+    dynamoDbClientConfig?: DynamoDBClientConfig,
     options?: MhDynamoClientOptions
   ) {
+    if (!dynamoDbClientConfig) dynamoDbClientConfig = {};
+
     const client = new DynamoDB(dynamoDbClientConfig);
```

Comment:
could get rid of the if:
`const client = new DynamoDB(dynamoDbClientConfig || {});`